enc/encode_parallel.cc

```diff
@@ -246,43 +272,83 @@ int BrotliCompressBufferParallel(BrotliParams params,
   } else if (params.lgblock > kMaxInputBlockBits) {
     params.lgblock = kMaxInputBlockBits;
   }
-  size_t max_input_block_size = 1 << params.lgblock;
 
-  std::vector<std::vector<uint8_t> > compressed_pieces;
+  // Calc number of blocks which will be distributed to threads
+  size_t max_input_block_size = 1 << params.lgblock;
+  size_t max_num_blocks       = std::max(size_t(1), (input_size / max_input_block_size));
+  std::vector<std::thread> threads;
+  
+  // Output of compression
+  std::vector<size_t> out_sizes(max_num_blocks);
+  std::vector<std::vector<uint8_t>> outs(max_num_blocks);
 
+  // Debug output
+  // std::cout << "max_num_blocks: " << max_num_blocks << std::endl;
+  // std::cout << "input_size: " << input_size << std::endl;
+  // std::cout << "max_input_block_size: " << max_input_block_size << std::endl;      
+  
   // Compress block-by-block independently.
-  for (size_t pos = 0; pos < input_size; ) {
-    size_t input_block_size = std::min(max_input_block_size, input_size - pos);
-    size_t out_size = input_block_size + (input_block_size >> 3) + 1024;
-    std::vector<uint8_t> out(out_size);
-    if (!WriteMetaBlockParallel(params,
-                                input_block_size,
-                                &input_buffer[pos],
-                                pos,
-                                input_buffer,
-                                pos == 0,
-                                pos + input_block_size == input_size,
-                                &out_size,
-                                &out[0])) {
-      return false;
-    }
-    out.resize(out_size);
-    compressed_pieces.push_back(out);
-    pos += input_block_size;
+  for(size_t thread_i = 0; thread_i < params.n_threads; thread_i++){
+  
+      std::thread t([&, thread_i](){
+
+	      size_t min_block_i    = 0;
+	      size_t max_block_i    = 0;
+	      std::tie(min_block_i, max_block_i) = consecutiveMapping(thread_i, params.n_threads, max_num_blocks);
+  
+	      for(size_t block_i = min_block_i; block_i < max_block_i; ++block_i){
+		  //std::cout << "[" <<  thread_i << "/"<< params.n_threads << "/" << block_i <<"] Compress block " << std::endl;
+      
+		  size_t input_block_size = std::min(max_input_block_size, input_size - ((block_i+1) * max_input_block_size));
+		  size_t pos              = block_i * max_input_block_size;
+		  out_sizes[block_i]      = 1.2 * input_block_size + 1024;
```

Comment:
You mixed `double`, `size_t` and `int` operation.
The original version was `size_t out_size = input_block_size + (input_block_size >> 3) + 1024` without a magic `1.2`